backend/capellacollab/tools/routes.py

```diff
@@ -2,31 +2,315 @@
+    try:
+        return crud.get_version_for_tool(tool_id, version_id, db)
+    except sqlalchemy.exc.NoResultFound:
+        raise HTTPException(
+            404,
+            {
+                "reason": f"The version with tool_id {tool_id} and version_id {version_id} was not found.",
+                "technical": f"Database returned 'None' when searching for the version with tool_id {tool_id} and version_id {version_id}.",
+            },
+        )
+
+
+def get_exisiting_tool_type(
+    tool_id: str, type_id, db: Session = Depends(get_db)
+) -> Type:
+    try:
+        return crud.get_type_for_tool(tool_id, type_id, db)
+    except sqlalchemy.exc.NoResultFound:
+        raise HTTPException(
+            404,
+            {
+                "reason": f"The version with tool_id {tool_id} and type_id {type_id} was not found.",
+                "technical": f"Database returned 'None' when searching for the version with tool_id {tool_id} and type_id {type_id}.",
+            },
+        )
+
+
+@router.get(
+    "",
+    response_model=list[ToolBase],
+    dependencies=[Depends(RoleVerification(required_role=Role.USER))],
+)
 def get_tools(db: Session = Depends(get_db)):
     return crud.get_all_tools(db)
 
 
 @router.get(
-    "/{tool_id}/versions", tags=["Tools"], response_model=list[ToolVersionBase]
+    "/{tool_id}",
+    dependencies=[Depends(RoleVerification(required_role=Role.USER))],
+)
+def get_tool_by_id(tool=Depends(get_existing_tool)):
+    return tool
+
+
+@router.post(
+    "",
+    response_model=ToolTypeBase,
+    dependencies=[Depends(RoleVerification(required_role=Role.ADMIN))],
+)
+def create_tool(body: CreateTool, db: Session = Depends(get_db)):
+    return crud.create_tool(db, models.Tool(name=body.name))
+
+
+@router.put(
+    "/{tool_id}",
+    response_model=ToolTypeBase,
+    dependencies=[Depends(RoleVerification(required_role=Role.ADMIN))],
+)
+def update_tool(tool_id: int, body: CreateTool, db: Session = Depends(get_db)):
+    tool = crud.get_tool_by_id(tool_id, db)
+    return crud.update_tool(db, tool, body)
+
+
+@router.delete(
+    "/{tool_id}",
+    status_code=204,
+    dependencies=[Depends(RoleVerification(required_role=Role.ADMIN))],
+)
+def delete_tool(
+    tool: Tool = Depends(get_existing_tool), db: Session = Depends(get_db)
+):
+    if tool.id == 1:
+        raise HTTPException(
+            403,
+            {
+                "reason": "The tool 'Capella' can not be deleted.",
+            },
+        )
+    try:
+        return crud.delete_tool(db, tool)
+    except sqlalchemy.exc.IntegrityError:
+        db.rollback()
+
+        dependencies = []
+        # Search for occurrences in project-models
+        for model in projects_models_crud.get_models_by_tool(tool.id, db):
+            dependencies.append(
+                f"Model '{model.name}' in project '{model.project.name}'"
+            )
+
+        for i in range(len(dependencies) - 1):
+            dependencies[i] = dependencies[i] + ","
```

Comment:
I'd keep db.rollback() in the route, because it is just one line (and we can leave it out of the function name then). I'll move the rest into a route `find_tool_dependencies()`